pkg/sql/pg_metadata_test.go

```diff
@@ -1125,16 +1123,22 @@ func (v *pgCatalogCodeVisitor) Visit(node ast.Node) ast.Visitor {
 		case virtualTablePopulateField:
 			// FuncLit is a function definition, we can look for addRow if this is
 			// a function definition.
-			val, ok := n.Value.(*ast.FuncLit)
-			if !ok {
+			switch val := n.Value.(type) {
+			case *ast.FuncLit:
+				v.bodyStmts = len(val.Body.List)
+				next, index := v.findAddRowFuncName(val.Type)
+				if index <= -1 {
+					v.pgCode.t.Fatal("populate function does not have a parameter with 'func(...tree.Datum) error' signature")
+				}
+				return next
+			case *ast.Ident:
+				if val.Name == "emptyVirtualTable" {
```

Comment:
@rafiss I'm not quite following everything this tool does. The change here allows the test to pass, but I'm not clear if there is other code that can be cleaned up. Rather than looking for a populate function whose body only contains `return nil`, we can now look for one with the name `emptyVirtualTable`. But will that allow rewriting to be done correctly?